File: Python/성격_유형_검사하기.py
def solution(survey, choices):
    result = dict(zip(["RT", "CF", "JM", 'AN'], [0, 0, 0, 0]))
    score = [[''.join(sorted(s)), -c] if s not in result else [s, c] for s, c in zip(survey, list(map(lambda x: x-4, choices)))]
    for s in score:
        result[s[0]] += s[1]
    return ''.join([r[0][0] if r[1] <= 0 else r[0][1] for r in result.items()])

File: Swift/성격_유형_검사하기.swift
import Foundation

func solution(_ survey:[String], _ choices:[Int]) -> String {
    var dict: [Character:Int] = ["R":0,"T":0,"C":0,"F":0,"J":0,"M":0,"A":0,"N":0]
    for i in survey.indices {
        let input = Array(survey[i])
        dict[input[choices[i] < 4 ? 0 : 1]]! += abs(4-choices[i])
    }
    var res = dict["R"]! >= dict["T"]! ? "R" : "T"
    res += dict["C"]! >= dict["F"]! ? "C" : "F"
    res += dict["J"]! >= dict["M"]! ? "J" : "M"
    res += dict["A"]! >= dict["N"]! ? "A" : "N"
    return res
}
